crates/aiken-lang/src/tipo/expr.rs

```diff
@@ -105,14 +109,20 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
 
         let mut arguments = Vec::new();
 
-        for (tipo, arg) in args_types.iter_mut().zip(args) {
+        for (index, (tipo, arg)) in args_types.iter_mut().zip(args).enumerate() {
             let CallArg {
                 label,
                 value,
                 location,
             } = arg;
 
-            let value = self.infer_call_argument(value, tipo.clone())?;
+            let value = self.infer_call_argument(value, tipo.clone());
+
+            let value = if index == 0 {
+                value.map_err(map_err)?
+            } else {
+                value?
+            };
```

Comment:
Why is this correct 🤔 ? I can't make sense of it.

Reply:
It's really confusing but I wrote tests to prove it. Essentially the pipe de sugars into function calls and if the first argument is wrong that means it's cause the pipe is wrong otherwise it's just a regular unification error.